### PR TITLE
fix to collections highlight bugs

### PR DESCRIPTION
**Pull request by abaevbog**
1. Trigger ZoteroPane.handleBlur on focusout instead of blur to remove highlight when itemTree entries loose focus even if ctrl/alt is still being held.
2. Prevent default event on Alt click on linux to avoid moving focus to menubar
3. Avoid re-triggering focus multiple times when ctrl is being held on windows to prevent highlight from becoming stuck and not going away.
4. Clear the highlight in `handleKeyUp` regardless of where the event comes from - this prevents highlight from being stuck when on windows focus is moved from itemTree before releasing `Ctrl`.

Fixes: #3410

### CODE REVIEW

File: chrome/content/zotero/zoteroPane.js [hunk 1]
```diff
@@ -1044,8 +1056,7 @@ var ZoteroPane = new function()
 	}
 	
 	function handleKeyUp(event) {
-		var from = event.originalTarget.id;
-		if (ZoteroPane.itemsView && from == ZoteroPane.itemsView.id) {
+		if (ZoteroPane.itemsView) {
```
Comment (dstillman) [inline chrome/content/zotero/zoteroPane.js:1059]: Probably don't need this at all, right? This was only there because we used it in the comparison.

Comment (abaevbog) [reply chrome/content/zotero/zoteroPane.js]: Oh yeah, that's right. I just force-pushed a commit without it

**Comment by abaevbog** [comment]
@adomasven, the glitch this was dealing with was not that you couldn't hold Alt and navigate items with arrows. The issues was that if you were to just click Alt to highlight collections for an item, release it, and then press ArrowDown, instead of navigating the itemTree to the next item, you would see the File menu would pop up and focus would jump to it. But, yeah, I see this certainly messes with default shortcuts.

I'm trying to see now if there's any way to not block the Alt event while still not having the menu pop up. A simpler solution, as you pointed out in #3684, would be to just use Ctrl instead of Alt. Is there any reason to not just do that and have Ctrl highlight collections on Linux, as well as windows?

**Comment by dstillman** [comment]
It's totally possible that we're using Alt on Linux solely because we were using Option on macOS, and that we changed Windows to Ctrl to avoid menu interactions without thinking about Linux. So we can probably switch to Ctrl on Linux if it avoids other problems. It would mean that everyone who has learned this secret key would have to learn another secret key, though, so we should only do it if it's actually a problem.

**Comment by abaevbog** [comment]
So, after some experimentation, my understanding is that after Alt is pressed, _sometimes_ we will want the menus to be activated to underline access keys (the next button click will navigate the menus), and _some other times_ we want the focus to remain on itemTree (when one highlights collections, so the next button click should navigate the itemTree). I feel like we do have to pick one or the other for everything to be consistent and robust. 

I did try to see if it's possible to not change the collection highlighting shortcut. If itemTree is focused, there's no way to tell if the user is after access keys or after collection highlights, so we could make a guess what to do with the menus/focus based on how long the key was held? E.g. more than X milliseconds - force the focus to remain on itemTree, otherwise - let it go to the menu. That is 100% janky and probably not that useful, so I'll get started on setting up Ctrl to be collection highlight trigger to free up Alt for the menu - but correct me if I'm wrong

**Comment by abaevbog** [comment]
Added #3689 to try to trigger focus on linux on Control instead of Alt

**Pull request merged**